pkg/handlers/update.go

```diff
@@ -420,3 +430,93 @@ func (h *Handler) GetAdminConsoleUpdateStatus(w http.ResponseWriter, r *http.Req
 	getAdminConsoleUpdateStatusResponse.Message = message
 	JSON(w, http.StatusOK, getAdminConsoleUpdateStatusResponse)
 }
+
+func (h *Handler) UploadAirgapBundle(w http.ResponseWriter, r *http.Request) {
+	appSlug := mux.Vars(r)["appSlug"]
+
+	app, err := store.GetStore().GetAppFromSlug(appSlug)
+	if err != nil {
+		logger.Error(errors.Wrapf(err, "failed to get app for slug %q", appSlug))
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	contentType := strings.Split(r.Header.Get("Content-Type"), ";")[0]
+	contentType = strings.TrimSpace(contentType)
+
+	if contentType != "multipart/form-data" {
+		logger.Error(errors.Errorf("unsupported content type: %s", r.Header.Get("Content-Type")))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if !app.IsAirgap {
+		logger.Error(errors.New("not an airgap app"))
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("Cannot update an online install using an airgap bundle"))
+		return
+	}
+
+	formReader, err := r.MultipartReader()
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to get multipart reader"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	foundAirgapBundle := false
+	for {
+		part, err := formReader.NextPart()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			logger.Error(errors.Wrap(err, "failed to get next part"))
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		if part.FormName() != "application.airgap" {
+			continue
+		}
+
+		foundAirgapBundle = true
+
+		destDir := filepath.Join(updatechecker.AvailableUpdatesPath, uuid.New().String())
```

Comment:
As it is now, you could theoretically have the same bundle uploaded multiple times and KOTS will not error. We could potentially address this by making the destination path deterministic based on channel ID and sequence or by KOTS storing a map of the currently available versions.